### DISCUSSION TITLE
ESP32S2 default method with generic method names like Neo800KbsMethod

### DISCUSSION THREAD

**Discussion by Makuna** [Makuna/NeoPixelBus#467] [Q&A] [answered]
@rahmanshaber; you asked about this on a closed issue.  I found that it wasn't being set whithout the RMT default symbol; so this got broken when I switched it back to I2S method being the default.  I checked in a fix for that yesterday.  For the ESP32S2, the default will always be RMT channel 3.  If you want to use I2S, only channel 0 is available and you can use the specific method names for that; like NeoEsp32I2s0Ws2812xMethod.

**Comment by rahmanshaber** [comment]
I have no idea what are you talking about. are you saying that now the example will build fine?

**Comment by Makuna** [comment, marked as answer]
Background Reading:  [NeoPixelBus Wiki on ESP32 Methods](https://github.com/Makuna/NeoPixelBus/wiki/ESP32-NeoMethods)

It is fixed in the Github repo.  You will need to remove the library from Arduino IDE and then [follow these instructions to get repo version](https://github.com/Makuna/NeoPixelBus#installing-this-library-from-github-advanced-you-want-to-contribute)

The fix will be in the next release.

OR you can just replace the Method with a specific methods mentioned in the background reading above.

**Comment by blazoncek** [comment]
@Makuna will the fix also solve this kind of errors?

.pio/libdeps/esp32-s2/NeoPixelBus/src/internal/NeoEsp32I2sMethod.h:327:9: error: 'NeoEsp32I2s1Ws2812xMethod' does not name a type; did you mean 'NeoEsp32I2s0Ws2812xMethod'?
.pio/libdeps/esp32-s2/NeoPixelBus/src/internal/NeoEsp32I2sMethod.h:330:9: error: 'NeoEsp32I2s1Ws2812xMethod' does not name a type; did you mean 'NeoEsp32I2s0Ws2812xMethod'?

**Reply by Makuna** [reply]
It should have fixed that.  The S2 doesn't have a second I2s (I2s1).  
Now note, if you are not using Arduino IDE to build with, you need to check your build to make sure it is correctly setting the right symbol so that the code knows its an S2.

**Reply by blazoncek** [reply]
That would be CONFIG_IDF_TARGET_ESP32S2 I guess?

**Reply by Makuna** [reply]
Correct, but note that you need the latest GITHUB version, which if the error you show above is correct, you do not have yet as the line numbers don't match up.

**Reply by blazoncek** [reply]
Thanks.
Indeed we use an older (release) version. I hope we'll be able to update NPB soon, since it is part of the libdeps.
